packages/fluentui/CONTRIBUTING.md

```diff
@@ -140,6 +140,43 @@ Please always use [`lerna add`][16] to manage all dependencies including interna
 lerna add @fluentui/react-proptypes packages/fluentui/react-northstar
 ```
 
+## Development Notes
+I (@rymeskar) have been part of the FluentUI framework team for the past two months. During these months, I have tried to collect ideas, rules, guidelines and best practices that are not otherwise captured. Generally, the recommendation is that code and conformant test is the best way to learn up-to-date FluentUI idioms.
+
+### Collection of Little Ideas
+* Changelog has a special format.
+* Always try to find the most relevant and similar component to get inspired by.
+* Try to extract [pure functions](https://en.wikipedia.org/wiki/Pure_function) outside of component's body; possibly into special files.
+* When developing date-time-utilities, one must run `yarn change` in the root and then  `yarn update-api` in the date-time-utilities folder.
+* Don't name boolean flags with 'is' prefix.
+* `Yarn build` is not necessary when working with FluentUI. You should just call `yarn` (resolve packages) and then `yarn start` (start application).
+* If you have issues with CI there is a [channel](https://teams.microsoft.com/l/channel/19%3a5b644df17ad64b89a6b7947969382baa%40thread.skype/Engineering%2520Systems?groupId=64c2bd5c-33c2-4898-918f-4a02a6ac98e1&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) to report such issues.
+* For benchmarking javaScript, one can use the https://jsperf.com .
+* The `useAutoControlled` hook description:
+  * `defaultValue` comes from an outer scope and be defined by user, i.e. `defaultChecked`.
+  * `initialValue` is internal and will be used if there is no `defaultValue` or `value`.
+  * `value` should come from outer space and always wins.
+
+
+### Collection of Concrete Topics
+* Testing
+  * For majority of components, visual test in combination with `isConformant` test is enough.
+  * Visual tests are based off examples. You can extend the examples by also providing 'steps' action tests.
+* Props
+  * Props should be a flattened object.
+  * Props should take advantage of optional properties in combination with defaultValues.
+* Event Handlers
+  * Going forward, we should use `onChange?: (e: React.SOME_EVENT<EL_TYPE>, data: DatepickerProps & { value: IDay }>)`. The currently used ComponentEventHandler uses React.SyntheticEvent and is able to reason about the proper type then.
+  * We use lodash's `_.invoke()`, the function ensures that the prop actually exists.
+  * We should work carefully with `event.preventDefault()`. Only invoke in necessary cases ([Example]( https://github.com/Semantic-Org/Semantic-UI-React/pull/4041)).
```

Comment:
Please could you add a case from Fluent UI? ;)

Reply:
Fixed.